pkg/codegen/schema/schema.go

```diff
@@ -535,6 +535,8 @@ type Function struct {
 	Outputs *ObjectType
 	// The return type of the function, if any.
 	ReturnType Type
+	// The return type is plain and not wrapped in an Output. This is only supported for resources.
+	ReturnTypePlain bool
```

Comment:
This feels really sad. I guess all of codegen is written to assume that Resource types are always in Outputs so you can't mark the Type as plain?

Reply:
Yeah. I would not say it's impossible to get there but it's certainly much less surgical. 

I'm less worried about editing codegen assumptions and more worried about editing per-language SDKs and serialization machinery that definitely assumes Resources have bags of outputs inside. I can't say I've explored this angle very deeply yet. Perhaps there's a way to get there in subsequent change.

Maybe there's a way to simply extend the code generation to add getters for them without interacting with the SDK machinery.

Reply:
Like something along these lines could be a pure-codegen change:

```typescript
export class WorkforcePool extends pulumi.CustomResource {
    public readonly someResourceProperty!: pulumi.Output<Resource | undefined>;

    // add this
    function getSomeResourceProperty(): Promise<Resource> {..}
}
```

We add new class members. It's not entirely aesthetically satisfying. If we don't do it and edit the type of someResourceProperty then the generic machinery for setting it has to be likewise carefully edited.

Reply:
Yeh I don't see any simple way to clean this up, something to try and come back to later I guess.

Reply:
Ah perhaps I misunderstood if you're wondering whether we can support this for methods that return something like `Output<string>` to return a `Promise<string>`. I think his may be considered, this restriction can be lifted here. The current code as written can probably force `Output<string>` just as well to a `Promise<string>`.

As with any type the questions are what to do with unknown, secret bit, and dependencies.

We currently blame the provider for unknown and that is likely the right approach for every type.

For secret bits and dependencies, I was thinking that we can "inject" them into the Resource somehow in the future, if we only guarantee this functionality for resources. That being said I didn't find an easy way to do this.

Our other options are for secret bits and non-empty dependencies apply equally well to Resource or String etc.:

- fail
- warn
- warn very quietly (log at -v=9)

Reply:
I'll think about this a bit and let other reviewers pipe in.

Reply:
>We currently blame the provider for unknown and that is likely the right approach for every type.

Yeh that feels right. 
Was more just that we need this extra flag for plain resources, but plain strings are just encoded in ReturnType.

Reply:
I've merged this, revising the code, this should support all values now not just resources.